android/app/src/main/java/org/fossasia/openevent/viewmodels/SponsorsFragmentViewModel.java

```diff
@@ -0,0 +1,38 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Sponsor;
+import org.fossasia.openevent.dbutils.LiveRealmData;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.List;
+
+public class SponsorsFragmentViewModel extends ViewModel {
+
+    private LiveData<List<Sponsor>> sponsorsList;
+    private RealmDataRepository realmRepo;
+
+    public SponsorsFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        sponsorsList = new MutableLiveData<>();
+        subscribeToSponsors();
+    }
+
+    public LiveData<List<Sponsor>> getSponsors() {
+        return sponsorsList;
+    }
+
+    private void subscribeToSponsors() {
+        LiveRealmData<Sponsor> sponsorLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getSponsors());
+        sponsorsList = Transformations.map(sponsorLiveRealmData, input -> input);
+    }
```

Comment:
This is not the correct way, you are assigning the variable twice. In `getSponsors` just check if the instance is null, if it is, then initialize the sponsorlist as you are doing here, or else just return

Please also create an issue for fixing this behaviour elsewhere

Reply:
sure :+1: